ecc/bls12-377/twistededwards/eddsa/marshal.go

```diff
@@ -118,9 +134,33 @@ func (sig *Signature) Bytes() []byte {
 // It returns the number of bytes read from buf.
 func (sig *Signature) SetBytes(buf []byte) (int, error) {
 	n := 0
-	if len(buf) < sizeSignature {
-		return n, io.ErrShortBuffer
+	if len(buf) != sizeSignature {
+		return n, ErrWrongSizeBuffer
```

Comment:
You can also do `return n, fmt.Errorf("wrong size buffer")`. In general, defining a error instance and returning it makes sense if we want to compare errors `if err == ErrWrongSizeBuffer` or `errors.Is(err, ErrWrongSizeBuffer)`. Otherwise, imo it makes the codebase unnecesarily complex (for example if we want to grep where the error was thrown).

Reply:
I ended up reintegrating the variables storing the errors for the tests :/